presto-common/src/main/java/com/facebook/presto/common/type/TimestampMicrosUtils.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.common.type;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.facebook.presto.common.type.SqlTimestamp.JSON_FORMATTER;
+import static com.facebook.presto.common.type.TimeZoneKey.UTC_KEY;
+import static java.lang.Math.floorDiv;
+
+public final class TimestampMicrosUtils
+{
+    public static final Pattern DATETIME_PATTERN = Pattern.compile("" +
+            "(?<year>\\d\\d\\d\\d)-(?<month>\\d{1,2})-(?<day>\\d{1,2})" +
+            "(?: (?<hour>\\d{1,2}):(?<minute>\\d{1,2})(?::(?<second>\\d{1,2})(?:\\.(?<fraction>\\d+))?)?)?" +
+            "\\s*(?<timezone>.+)?");
+
+    public static final long MICROS_PER_SECOND = 1000_000;
+    public static final long MICROS_PER_MILLI = 1000;
+    public static final long NANOS_PER_MICRO = 1000;
+    public static final int PRECISION_MICROSECONDS = 6;
+
+    private static final long[] POWERS_OF_TEN = {
+            100_000L,
+            10_000L,
+            1_000L,
+            100L,
+            10L,
+            1L
+    };
+
+    private TimestampMicrosUtils() {}
+
+    public static double unixTimeToMicros(double unixTime)
+    {
+        return unixTime * MICROS_PER_SECOND;
```

Comment:
Any particular reason not to use java.util.concurrent.TimeUnit ?

Reply:
TimeUnit can only be used by a few methods, such as unixTimeToMicros / microsToUnixTime / microsToMillis / millisToMicros.
TimeUnit is not used here for format purposes.